### DISCUSSION TITLE
Next.js missing cache-control header may lead to CDN caching empty reply

### DISCUSSION THREAD

**Discussion by saurishkar** [vercel/next.js#59009] [Help]
#### Issue Type
Security

#### Issue description
This particular [issue](https://github.com/advisories/GHSA-c59h-r6p8-q9wc) has been raised last month and has been fixed [here](https://github.com/vercel/next.js/pull/54732) in the canary branch.

Wanted to check if it is possible to raise a patch version for the fix in v12 since it is being flagged in version 12 as well.
If the above isn't feasible, can someone suggest how to fix this for next@12 ?

**Comment by saurishkar** [comment]
Sorry to bother you @remorses but wanted to get your perspective since you raised the fix for the same.

**Reply by remorses** [reply]
Why can't you upgrade Next?

**Reply by saurishkar** [reply]
@remorses 
The project I am working on is a monorepo with multiple teams working on different areas of the project. Upgrading to Next 13 will be a difficult task which will stretch to more than a couple of days that can disrupt their current sprints.

**Reply by saurishkar** [reply]
@remorses can you help me find the fix in v12 so that I can fork the repo as long as we are not moving it to v13?

**Comment by trevorwright** [comment]
Is there any progress on this conversation? Our team is in a similar position of a monorepo which still has some outstanding efforts to successfully migrate to React 18 before we can upgrade beyond Next 12. The fix from v13+ does not seem to be easily patchable into the v12 server-base but happy to have someone with more experience tell me I'm wrong on this one.
